DQMServices/Core/src/DQMStore.cc

```diff
@@ -405,14 +405,14 @@ void DQMStore::mergeAndResetMEsRunSummaryCache(uint32_t run,
 	{
 	  if(me->getTH1()->CanExtendAllAxes() && i->getTH1()->CanExtendAllAxes()) {
 	    TList list;
-	    list.Add(i->getTH1());
+	    list.Add(i->getTH1().operator->().operator->());
```

Comment:
I did a test a few days back, for this specific case, and it works "properly". The lock is cleared **after** return from Add(), at least with gcc.

But why not simply add a `MonitorElement::exec(std::function<void(TH1 *)>)`.
And then call it with:
```
t.exec([&](TH1* x) mutable {
    TList list;
    list.Add(x);
    if( -1 == me->getTH1()->Merge(&list))...
})
```
I think this lambda simplifies the code in a few places, you can also chain this lambda to lock 2 or more elements...

mergeAndResetMEsRunSummaryCache() must lock! Previously, without locking, there was a guarantee that mergeAndReset* methods only accessed local-to-module histograms. This is no longer the case....